### PR TITLE
Bypass base64 encode and decode

### PR DESCRIPTION
**Pull request by michaelwoods**
We observed that the export of long recordings (8min) required several minutes to export into a Blob. This is seemingly due to the data being encoded into base64 and then immediately decoded.

This change bypasses the base64 translations.

### CODE REVIEW

File: src/main/flashwavrecorder/RecorderJSInterface.as [hunk 1]
```diff
@@ -250,6 +251,16 @@ package flashwavrecorder {
...
     }
 
+    private function getWAV(name:String):ByteArray {
+      var data:ByteArray;
+      try {
+        data = _recorder.convertToWav(name);
+      } catch (e:Error) {
+        data = new ByteArray();
+      }
+      return data;
```
Comment (michalstocki) [inline src/main/flashwavrecorder/RecorderJSInterface.as:261]: Wow. It seems that we can return a ByteArray object through an external interface? What type it became on the javascript side?

File: html/js/recorder.js [hunk 2]
```diff
@@ -109,12 +109,11 @@
     },
 
     getBlob: function(name) {
-      var base64Data = Recorder.getBase64(name).split(',')[1];
-      return base64toBlob(base64Data, RECORDED_AUDIO_TYPE);
+      return Recorder.recorder.getWAV(name);
```
Comment (michalstocki) [inline html/js/recorder.js:112]: I want to be sure that this function returns the [`Blob`](https://developer.mozilla.org/pl/docs/Web/API/Blob) object. Is it really a `Blob` instance?

File: html/js/recorder.js [hunk 3]
```diff
@@ -109,12 +109,11 @@
     },
 
     getBlob: function(name) {
-      var base64Data = Recorder.getBase64(name).split(',')[1];
-      return base64toBlob(base64Data, RECORDED_AUDIO_TYPE);
```
Comment (michalstocki) [inline html/js/recorder.js:113]: We could drop the `base64toBlob` function if it's no longer needed.

File: html/js/recorder.js (hunk 2 above)
Comment (michalstocki) [reply html/js/recorder.js:112]: I've just checked this and it seems that now, `getBlob(name)` returns simply plain javascript object without any data:

```
    {
        bytesAvailable: 0,
        endian: "littleEndian",
        length: 92204,
        objectEncoding: 3,
        position: 92204,
        shareable: false
    }
```

File: src/main/flashwavrecorder/RecorderJSInterface.as (hunk 1 above)
Comment (michalstocki) [reply src/main/flashwavrecorder/RecorderJSInterface.as:261]: Unfortunately that will not work. It seem that there is no way to send binary data between flash and javascript. All data need to be converted to string. I found some questions on SO on this topic:
http://stackoverflow.com/q/22952032/1346003
http://stackoverflow.com/a/11811097/1346003

**Review by michalstocki — CHANGES_REQUESTED** [review]
The idea of bypassing encoding and decoding base64 to transfer binary data from Flash to JavaScript is good. Unfortunately I can't get your solution to work.

**Comment by michaelwoods** [comment]
Unfortunately I only tested this against a mock server and hadn't verified the output. It seems that you're correct, we won't be able to pass binary data across the interface. The answer to "why is it encoding and decoding?" is now clear.

I will have to explore another solution. Thanks for the review.

**Pull request closed without merging**